### PR TITLE
Added asin function and test for frontend

### LINKED ISSUE #15799: asin

### PR DESCRIPTION
**Pull request by VladZetta**
close #15799

### CODE REVIEW

**Comment by VladZetta** [comment]
Hi @hello-fri-end !
Could you please review my PR? I get some checks failed ( 1 for intelligent-test-pr(6) and 1 for formatting). I just cannot find an error.  Sorry for running too much test, but I really want to find the error :)

Thanks!

File: ivy/functional/frontends/paddle/tensor/math.py [hunk 1]
```diff
@@ -36,7 +36,13 @@ def tanh(x, name=None):
     return ivy.tanh(x)
 
 
-@with_unsupported_dtypes({"2.4.2 and below": ("float16", "bfloat16")}, "paddle")
```
Comment (hello-fri-end) [inline ivy/functional/frontends/paddle/tensor/math.py:39]: Why did you remove this?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py [hunk 2]
```diff
@@ -164,3 +164,30 @@ def test_paddle_acosh(
         atol=1e-2,
         x=x[0],
     )
+
+
+# asin
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.asin",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py:173]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```

Can you make this change and let me know if you face any issues?

**Review by hello-fri-end — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/paddle/tensor/math.py (hunk 1 above)
Comment (VladZetta) [reply ivy/functional/frontends/paddle/tensor/math.py:39]: That's a mistake, didn't see that

**Review by VladZetta — COMMENTED** [review]
Changed Dtypes

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py (hunk 2 above)
Comment (VladZetta) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_math.py]: @hello-fri-end ,I made the changes , now two test failing intelligent-tests-pr / run_tests (1 and 6)

**Comment by VladZetta** [comment]
> > Hi @hello-fri-end ! Could you please review my PR? I get some checks failed ( 1 for intelligent-test-pr(6) and 1 for formatting). I just cannot find an error. Sorry for running too much test, but I really want to find the error :)
> > Thanks!
> 
> Hey @VladZetta ! The documentation of `paddle.asin` seems to be wrong as it's throwing an error when `float16` is used. You can confirm the same by running the following code:
> 
> ```python
> dtypes_list = ["int8", "int16", "int32", "int64", "uint8", "float16", "float32", "float64", "complex64", "complex128", "bool",]
> unsupported = []
> supported = []
> for dtype in dtypes_list:
>     try:
>         a = paddle.to_tensor([1.,2,3],dtype=dtype)
>         # insert the function you want to test in the following line
>         #paddle.normal(a,a)
>         paddle.asin(a)
>         supported.append(a.dtype)
>     except:
>         # use this to generate native dtype
>         unsupported.append(a.dtype)
>         # use this to generate string dtype
>         #unsupported.append(dtype)
> print(unsupported)
> print(supported)
> ```
> 
> If you remove `float16` from the supported dtypes, I think the tests will pass. Let me know if you have any other questions :)
> 
> P.S: Could you please link the issue in the description that this PR closes.
@hello-fri-end 
Well, indeed float16 is not supported, changed it! and added link in description to the issue

But tests are still failing. I don't get why you suggested using `get_dtypes("valid")`. Is there a way to make it return only `float32 and float64` ? I'm just guessing

**Review by hello-fri-end — APPROVED** [review]
The changes look good to me! The errors in the CI are unrelated to your PR. Thanks for the contribution :)

**Comment by VladZetta** [comment]
> The changes look good to me! The errors in the CI are unrelated to your PR. Thanks for the contribution :)

Thanks for helping out!

**Comment by hello-fri-end** [comment]
> > > Hi @hello-fri-end ! Could you please review my PR? I get some checks failed ( 1 for intelligent-test-pr(6) and 1 for formatting). I just cannot find an error. Sorry for running too much test, but I really want to find the error :)
> > > Thanks!
> > 
> > 
> > Hey @VladZetta ! The documentation of `paddle.asin` seems to be wrong as it's throwing an error when `float16` is used. You can confirm the same by running the following code:
> > ```python
> > dtypes_list = ["int8", "int16", "int32", "int64", "uint8", "float16", "float32", "float64", "complex64", "complex128", "bool",]
> > unsupported = []
> > supported = []
> > for dtype in dtypes_list:
> >     try:
> >         a = paddle.to_tensor([1.,2,3],dtype=dtype)
> >         # insert the function you want to test in the following line
> >         #paddle.normal(a,a)
> >         paddle.asin(a)
> >         supported.append(a.dtype)
> >     except:
> >         # use this to generate native dtype
> >         unsupported.append(a.dtype)
> >         # use this to generate string dtype
> >         #unsupported.append(dtype)
> > print(unsupported)
> > print(supported)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If you remove `float16` from the supported dtypes, I think the tests will pass. Let me know if you have any other questions :)
> > P.S: Could you please link the issue in the description that this PR closes.
> 
> @hello-fri-end Well, indeed float16 is not supported, changed it! and added link in description to the issue
> 
> But tests are still failing. I don't get why you suggested using `get_dtypes("valid")`. Is there a way to make it return only `float32 and float64` ? I'm just guessing

`get_dtypes("valid")` automatically finds all the dtypes that are supported by the function, so, even if the function only supports `floats`, that's what `valid` will return :)

**Comment by hello-fri-end** [comment]
add_frontend_checklist

**Comment by hello-fri-end** [comment]
> > The changes look good to me! The errors in the CI are unrelated to your PR. Thanks for the contribution :)
> 
> Thanks for helping out! So PR will be accepted? Because it seems like I need to send the link of PR to go further :)

Merged it :rocket:
